### PR TITLE
Multiple features added to make eval Smoother

### PR DESCRIPTION
**Pull request by fi5421**
## Description

bird_eval now logs errors
generate_batch_file now exits on error
vectorize now removes the target question from the shots if needed 
change_model reads batch input file to only change of the model to that in the config file
fine_tune file makes training examples from the current configuration in the config file

### CODE REVIEW

File: server/app/main.py
Comment (Mehak-Conrad) [inline server/app/main.py:1]: It should be Database name, I checked the request_schema.py and it is database_type there too. My bad, I did not see this correctly after merging the branch that did this with main. 

I updated all variable named database_type to database_name as it was more descriptive. The set_database itself is using database_name.

File: server/app/main.py [hunk 1]
```diff
@@ -261,7 +261,7 @@ async def generate_prompt(request: PromptGenerationRequest):
 @app.post("/database/change/")
 async def change_database(request: ChangeDatabaseRequest):
     try:
-        db.set_database(request.database_name)
+        db.set_database(request.database_type)
```
Comment (Mehak-Conrad) [inline server/app/main.py:264]: It should be Database name, I checked the request_schema.py and it is database_type there too. My bad, I did not see this correctly after merging the branch that did this with main.

I updated all variable named database_type to database_name as it was more descriptive. The set_database itself is using database_name.

File: server/scripts/download_batch_output.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 downloaded=[]
 
 # Enter the correct timestamp file of the last run batch jobs
-time_stamp="2024-11-04_20:53:24.txt"
+time_stamp="2024-11-25_12:00:55.txt"
```
Comment (AwaisKamran) [inline server/scripts/download_batch_output.py:17]: Why is this required?

File: server/scripts/fine_tune.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    GENERATE_BATCH_RELATIVE_PATH,
+    DB_CHANGE_ENPOINT,
+    PROMPT_GENERATE_ENDPOINT,
+    PROMPT_TYPE,
+    NUM_SHOTS
+)
+
+
+
+
```
Comment (AwaisKamran) [inline server/scripts/fine_tune.py:24]: Avoid adding too many trailing spaces when writing code, one space creates enough space

File: server/scripts/download_batch_output.py (hunk 2 above)
Comment (fi5421) [reply server/scripts/download_batch_output.py:17]: The scripts as a downloader of the correct batch files. I use the timestamp in a previous script to save the batch ids and their dbs. I use this timestamp in the downloader as an indentifier to download the correct batches.

File: server/app/main.py (hunk 1 above)
Comment (fi5421) [reply server/app/main.py:264]: Ok Ill update the request_schema in this PR as well.

**Review by Mehak-Conrad — APPROVED** [review]

File: server/app/db.py [hunk 4]
```diff
@@ -38,3 +38,4 @@ def get_db():
         yield db
     finally:
         db.close()
+
```
Comment (AwaisKamran) [inline server/app/db.py:41]: One space is enough space

File: server/scripts/download_batch_output.py (hunk 2 above)
Comment (AwaisKamran) [reply server/scripts/download_batch_output.py:17]: Why timestamps? Cant we just generate a UUID as a tracker

File: server/scripts/fine_tune.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+# iterating over all Databases to generate prompts and make batch input files
+for database in tqdm(directories,desc=f'Processing Directories:'):
+
+    response=requests.post(DB_CHANGE_ENPOINT,json={'database_type':database,"sample_path":f"{GENERATE_BATCH_RELATIVE_PATH}{database}{SAMPLE_QUESTIONS_DIR}unmasked_{database}.json"})
+
+    with open(f"{GENERATE_BATCH_SCRIPT_PATH}{database}{SAMPLE_QUESTIONS_DIR}unmasked_{database}.json") as file:
+        sample_qs=json.loads(file.read())
+        file.close()
+    
+    if not response.status_code==200:
```
Comment (AwaisKamran) [inline server/scripts/fine_tune.py:36]: Always use intermediate strings and number for the checks as constants declared at the top of the script

File: server/scripts/download_batch_output.py (hunk 2 above)
Comment (fi5421) [reply server/scripts/download_batch_output.py:17]: We could use a UUID but this way it's a little easier to see which batch was run when.
Also we used this convention in all bird experiments and the batch_metadata is stored against this convention for all experiments so I would not recommend changing the convention now.

**Review by AwaisKamran — APPROVED** [review]
